src/test/java/uk/gov/companieshouse/service/AppealServiceTest.java

```diff
@@ -0,0 +1,114 @@
+    public static final String TEST_REASON_DESCRIPTION = "This is a description";
+
+    @InjectMocks
+    private AppealService appealService;
+
+    @Mock
+    private AppealRepository appealRepository;
+
+    @Test
+    public void testCreateAppeal_returnsResourceId() throws Exception {
+
+        ArgumentCaptor<Appeal> appealArgumentCaptor = ArgumentCaptor.forClass(Appeal.class);
+
+        Appeal appeal = TestUtil.getValidAppeal();
+        appeal.setId(TEST_RESOURCE_ID);
+
+        when(appealRepository.insert(any(Appeal.class))).thenReturn(appeal);
+
+        appealService.saveAppeal(appeal, TEST_ERIC_ID);
+
+        verify(appealRepository).insert(appealArgumentCaptor.capture());
+
+        assertAll("Create appeal sets created by and created at",
+            () -> assertEquals(TEST_ERIC_ID, appealArgumentCaptor.getValue().getCreatedBy().getId()),
+            () -> assertNotNull(appealArgumentCaptor.getValue().getCreatedAt()));
+
+        assertAll("Create appeal returns resource id",
+            () -> assertNotNull(appealArgumentCaptor.getValue().getId()),
+            () -> assertEquals(TEST_RESOURCE_ID, appealArgumentCaptor.getValue().getId()));
+    }
+
+    @Test
+    public void testCreateAppeal_throwsExceptionIfNoResourceIdReturned() {
+
+        Appeal appeal = TestUtil.getValidAppeal();
+
+        when(appealRepository.insert(any(Appeal.class))).thenReturn(appeal);
+
+        assertThrows(Exception.class, () -> appealService.saveAppeal(appeal, TEST_ERIC_ID));
+    }
+
+    @Test
+    public void testCreateAppeal_throwsExceptionIfUnableToInsertData() {
+
+        Appeal appeal = TestUtil.getValidAppeal();
+
+        when(appealRepository.insert(any(Appeal.class))).thenReturn(null);
+
+        assertThrows(Exception.class, () -> appealService.saveAppeal(appeal, TEST_ERIC_ID));
+    }
+
+    @Test
+    public void testGetAppealById_returnsAppeal() {
+
+        Appeal validAppeal = TestUtil.getValidAppeal();
+
+        when(appealRepository.findById(any(String.class))).thenReturn(Optional.of(validAppeal));
+
+        Optional<Appeal> appealOpt = appealService.getAppeal(TEST_RESOURCE_ID);
+
+        assertTrue(appealOpt.isPresent());
+
+        Appeal appeal = appealOpt.get();
+
+        assertAll("Get appeal returns appeal with resource ID",
+            () -> assertEquals(TEST_PENALTY_REFERENCE, appeal.getPenaltyIdentifier().getPenaltyReference()),
+            () -> assertEquals(TEST_COMPANY_ID, appeal.getPenaltyIdentifier().getCompanyNumber()),
+            () -> assertEquals(TEST_REASON_TITLE, appeal.getReason().getOther().getTitle()),
+            () -> assertEquals(TEST_REASON_DESCRIPTION, appeal.getReason().getOther().getDescription()));
+    }
+
+    @Test
+    public void testGetAppealById_returnsEmptyAppeal() {
+
+        when(appealRepository.findById(any(String.class))).thenReturn(Optional.empty());
+
+        Optional<Appeal> appealOpt = appealService.getAppeal(TEST_RESOURCE_ID);
```

Comment:
For the future record I do not think it makes sense to call variables with `Opt` prefix like `appealOpt`. Variable still represents appeal just wrapped for convenience and by being type of variable I am being reassured that it is optional value.

Reply:
renamed to `appeal`